File: submissions/cpp/873.cpp
class Solution {
public:
    int lenLongestFibSubseq(vector<int> &A) {
        unordered_set<int> S;
        for (auto a:A)
            S.insert(a);
        int maxLen = 0;
        for (int i = 0; i < A.size(); ++i) {
            for (int j = i + 1; j < A.size(); ++j) {
                int a = A[i], b = A[j], &&len = 0;
                while (S.find(a + b) != S.end()) {
                    int c = b;
                    b = a + b;
                    a = c;
                    ++len;
                    maxLen = max(maxLen, len);
                }
            }
        }
        return maxLen == 0 ? 0 : maxLen + 2;
    }
};

File: LeetCode/python3/873.py
class Solution:
    def lenLongestFibSubseq(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        maxLen = 0
        S = set(A)
        for i in range(0, len(A)):
            for j in range(i + 1, len(A)):
                a, b = A[j], A[i] + A[j]
                cuurentLen = 2
                while b in S:
                    a, b = b, a + b
                    cuurentLen += 1
                maxLen = max(maxLen, cuurentLen)
        return maxLen if maxLen >= 3 else 0